kotlinx-coroutines-core/common/src/CoroutineScope.kt

```diff
@@ -10,102 +10,481 @@ import kotlin.coroutines.*
 import kotlin.coroutines.intrinsics.*
 
 /**
- * Defines a scope for new coroutines. Every **coroutine builder** (like [launch], [async], etc.)
- * is an extension on [CoroutineScope] and inherits its [coroutineContext][CoroutineScope.coroutineContext]
- * to automatically propagate all its elements and cancellation.
+ * A scope in which coroutines run.
  *
- * The best ways to obtain a standalone instance of the scope are [CoroutineScope()] and [MainScope()] factory functions,
- * taking care to cancel these coroutine scopes when they are no longer needed (see the section on custom usage below for
- * explanation and example).
+ * A scope defines a group of coroutines with shared execution properties
+ * (represented as [CoroutineContext.Element] values)
```

Comment:
```suggestion
 * A scope defines a group of coroutines with shared execution properties
 * (for example, on which threads the coroutines execute or which thread-local variables are tracked in them)
```
Would this clarify the meaning of "execution properties" here and for the rest of the kdoc?

Reply:
Late comment 🙈

There's nothing thread-local in a coroutine scope. `CoroutineContext.Element`s are an analogue, but are not thread-bound. Mentioning "thread-local" like this would create confusion with `ThreadLocal` in Java and equivalent in other platforms or libraries.

"are tracked in them" isn't clear to me either.